### DISCUSSION TITLE
Uptime format optional days

### DISCUSSION THREAD

**Discussion by omnigenous** [fastfetch-cli/fastfetch#1044] [Q&A] [answered]
`"format": "{2}h {3}m"`
Real uptime: 26 hours 34 minutes 17 seconds
`2h 34m`

then I changed format:
`"format": "{1}d {2}h {3}m"`
But I don't want to see `0d ` most of the time when my machine is not up for several days.

I want to make `{1}d ` part (including space) optional and only display when hours >= 24 or days >= 1.

I tried different combinations with question mark and brackets, none of them work:
`{1?}d {2}h {3}m` -> `{1?}d 2h 53m`
`{?1}d {2}h {3}m` -> `d 2h 53m`
`{{1}d ?}{2}h {3}m` -> `{1d ?}2h 54m`
`{?{1}d }{2}h {3}m` -> `{?1d }2h 54m`
`{{1}d ?}{2}h {3}m` -> `{1}d ?}2h 54m`
`{?{1}d }{2}h {3}m` -> `{?{1}d }2h 55m`

**Comment by CarterLi** [comment, marked as answer]
`fastfetch -s uptime --uptime-format '{?1}{1}d {?}{?2}{2}h {?}{3}m'`

**Reply by omnigenous** [reply]
Oh, I see now `{?CONDITION}IF TRUE{?}`

**Comment by MiniRobotDev** [comment]
is there anyway to pad the numbers :
format:     1h 2m 3s 
current:    1:2:3
expected: 01:02:03
i pad the numbers with 2 digit and fill missing digit with 0
what is the syntax to do that is there something like >0,2 exist ?
